### ISSUE TITLE
`tf_spread` default

### ISSUE THREAD

**Issue by jeff-goldsmith** [tidyfun/tidyfun#112]
should `tf_spread` default to `interpolate = TRUE`? for irregular data, this will fill in a bunch of missing values in a way that converting to a matrix doesn't -- seems a bit unexpected to me ...

**Comment by fabian-s** [comment]
true. 

- i'll reset the default to FALSE

- i would also like to reverse the *mandatory* `arg` for irregular tfd:
```
 if (missing(arg)) {
    if (is_irreg(tf)) {
      stop("need explicit <arg> for irregular ", sQuote(tf_var), " -- otherwise
           irregular grids can produce extremely wide dataframes.")
    }
    arg <- tf_arg(tf)
.....
 ```
 and replace it with
 ```
 if (missing(arg)) {
  arg <- tf_arg(tf)
  if (is_irreg(tf)) {
    arg <- unlist(arg) %>% unique() %>% sort() 
    warning("no explicit <arg> for irregular ", sQuote(tf_var), " provided-- using all ",
            length(arg), "distinct observed argument values.")
  }
 ```
i think it's better UX to warn about a *potentially* undesirable output for the default input instead of completely aborting something that might be the intended result and is the most direct translation of irregular data into a data.frame.

thoughts?

**Comment by fabian-s** [comment]
closed in #114
